cg/constants/demultiplexing.py

```diff
@@ -88,3 +88,10 @@ class DemultiplexingDirsAndFiles(StrEnum):
     RUN_PARAMETERS: str = "RunParameters.xml"
     SAMPLE_SHEET_FILE_NAME: str = "SampleSheet.csv"
     UNALIGNED_DIR_NAME: str = "Unaligned"
+
+
+class SampleSheetType(StrEnum):
+    """Define sample sheet type."""
+
+    S2: str = "S2"
+    S4: str = "S4"
```

Comment:
I lack the knowledge of which instrument uses each type. I ask the wisdom of @Karl-Svard @karlnyr in this matter!

Reply:
I talked to @Karl-Svard, and it seems to be more complicated than that,so better discuss it with them

Reply:
So I believe that we currently infer the type of sequencer from the parsed RunParameters file which is the "FlowCellMode". So to say that the type is S4, S2 is incorrect - the type should be the sequencer type. With the new sequencing machines, we are swapping flow cells to 5B, 10B, and 25B which most likely will be behind the same tag. To be honest, I think that this information is best fetched from LIMS/Arnold as I would rather rely on the setup steps to determine this rather than the output format of a file that Illumina designs. That would also remove some very "sketchy" xmltree coding which I find un-intuitive. I will however ask Emilia at Illumina about possible changes to any setup files/output files which could affect not only the sample sheet creation but also the parsing of demultiplexing data.

Reply:
From cgmodels:
```
sheet_type: Literal["2500", "SP", "S2", "S4"]
novaseqsample = {"bcl2fastq": NovaSeqSampleBcl2Fastq, "dragen": NovaSeqSampleDragen}
sample_type = Sample if sheet_type == "2500" else novaseqsample[bcl_converter]

class NovaSeqSampleBcl2Fastq(NovaSeqSample):
    sample_id: str = Field(..., alias="SampleID")
    project: str = Field(..., alias="Project")

class NovaSeqSample(Sample):
    """This model is used when parsing/validating existing novaseq sample sheets"""

    second_index: str = Field(..., alias="index2")

class SampleSheet(BaseModel):
    type: str
    samples: List[Sample]
```

It seems that the sheet type is coupled to the software on each sequencing instrument and how that instrument expects certain headers to be named. I guess certain flow cell types are able to be processed by more than one instrument, but it should be how the instrument parses the sheet that is the determining factor here.

@karlnyr @Karl-Svard Is it correct to assume that the sheet format per instrument does not change depending on flow cell type?